curated_transformers/tokenization/bbpe_tokenizer.py

```diff
@@ -0,0 +1,131 @@
+        """
+        self.unk_piece = unk_piece
+        self.bos_piece = bos_piece
+        self.eos_piece = eos_piece
+        self.pad_piece = pad_piece
+        self.processor = processor
+
+    @classmethod
+    def from_files(
+        cls: Type[Self],
+        *,
+        vocab_path: Path,
+        merges_path: Path,
+        bos_piece: str = "<s>",
+        eos_piece: str = "</s>",
+        pad_piece: str = "<pad>",
+        unk_piece: str = "<unk>",
+    ) -> Self:
+        """Construct a tokenizer from vocabulary and merge files.
+
+        vocab_path (Path): path to the vocabulary file.
+        merges_path (Path): path to the merges file.
+        bos_piece (str): The piece to use to mark the beginning of a sequence.
+        eos_piece (str): The piece to use to mark the end of a sequence.
+        pad_piece (str): Padding piece.
+        unk_piece (str): The piece to use to mark unknowns.
+        """
+        processor = ByteBPEProcessor.load_from_files(
+            vocab=vocab_path, merges=merges_path
+        )
+        return cls(
+            processor=processor,
+            bos_piece=bos_piece,
+            eos_piece=eos_piece,
+            pad_piece=pad_piece,
+            unk_piece=unk_piece,
+        )
+
+    def _tokenize(self, input: List[str]) -> PiecesWithIds:
+        bos_id = self.processor.piece_id(self.bos_piece)
+        if bos_id is None:
+            raise ValueError(
+                f"Byte BPE piece encoder vocabulary doesn't contain '{self.bos_piece}' piece"
+            )
+
+        eos_id = self.processor.piece_id(self.eos_piece)
+        if eos_id is None:
+            raise ValueError(
+                f"Byte BPE piece encoder vocabulary doesn't contain '{self.eos_piece}' piece"
+            )
+
+        pad_id = self.processor.piece_id(self.pad_piece)
+        if eos_id is None:
+            raise ValueError(
+                f"Byte BPE piece encoder vocabulary doesn't contain '{self.eos_piece}' piece"
+            )
+
+        ids = []
+        pieces = []
+        lens = []
+
+        for text in input:
+            text_ids = [bos_id]
+            text_pieces = [self.bos_piece]
+            text_lens = [1]
+
+            for idx, token in enumerate(text.split(" ")):
+                if idx != 0:
+                    token = " " + token
+                token_ids, token_pieces = self.processor.encode(token)
+                text_ids.extend(token_ids)
+                text_pieces.extend(token_pieces)
+                text_lens.append(len(token_ids))
+
+            text_ids.append(eos_id)
+            text_pieces.append(self.eos_piece)
+            text_lens.append(1)
+
+            ids.append(text_ids)
+            pieces.append(text_pieces)
+            lens.append(text_lens)
+
+        return PiecesWithIds(pad_id=pad_id, ids=ids, lens=lens, pieces=pieces)
+
+    @classmethod
+    def _convert_hf_tokenizer(cls: Type[Self], tokenizer) -> Self:
```

Comment:
```suggestion
    def _convert_hf_tokenizer(cls: Type[Self], tokenizer: Any) -> Self:
```
Or we go for the full annotation by importing `tokenizers`.